src/plugins/home/public/application/components/sample_data_set_card.js

```diff
@@ -65,6 +65,15 @@ export class SampleDataSetCard extends React.Component {
   };
 
   renderBtn = () => {
+    const dataSourceEnabled = this.props.isDataSourceEnabled;
+    const hideLocalCluster = this.props.isLocalClusterHidden;
+    const dataSourceId = this.props.dataSourceId;
+
+    let buttonDisabled = false;
+    if (dataSourceEnabled && hideLocalCluster) {
+      buttonDisabled = dataSourceId === undefined;
```

Comment:
Looks like we are using same checks as in saved object management page, can we extract the same check into one util function?

Reply:
Thanks for the comment.

That's a good concerns and I used to think of it.
I prefer to keep it separate since these UX components are in different plugins, it will be confused to put the judgment into a specific plugin's Util.

Reply:
Thanks for the response, replyed under this thread:
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5827#discussion_r1490161908